design/001-RPC-Interface-Specification.md

```diff
@@ -0,0 +1,413 @@
+    * 2e50w for currency "Ethereum" that represents 2 ethers and 50 wei.
+
+4. `Channel`
+
+    * `Channel ID`: [String] Unique ID of the channel.
+    * `Channel Balance`: [Balance Info]
+    * `Version`: [Integer] Current Version of the channel.
+
+The following error messages will be used in the API description.
+
+1. `Unknown Session ID`: No session corresponding to the specified ID.
+2. `Unknown Channel ID`: No channel corresponding to the specified ID.
+3. `Unknown Subscription ID`: No subscription corresponding to the specified ID.
+4. `Peer Exists`: A contact entry exits for the given peer and cannot be
+5. `Peer Not Responding`: Peer did not respond within expected time period.
+
+### Node
+
+#### 1. Get Config
+
+Returns the configuration parameters of the node.
+
+*Parameters* none
+
+*Return*
+
+* `Node config`: (list of configurable parameters is yet to be defined)
+
+#### 2. Open Session
+
+Open a new session for the given user with the specified configuration file.
+
+*Parameters*
+
+* `Config File`: [String] Path to the config file for the session. This should
+   present on a file system path accesible by the node.
+
+*Return*
+
+* `Session ID`: [String] Unique ID of the session.
+* `Success`: [Boolean]
+
+*Errors*
+
+* `Session already exists`
+
+#### 3. Help
+
+Returns the help message.
+
+*Parameters* none
+
+*Return*
+
+* `Help Message`: list of available commands.
+
+### Session
+
+#### 1. Add Contact
+
+Add a peer to the contacts in the specified session.
+
+*Parameters*
+
+* `Session ID`: [String] Unique ID of the session.
+* `Contact`: [Peer] Details of peer to be added to contacts.
+
+*Return*
+
+* `Success`: [Boolean]
+
+*Errors*
+
+* `Unknown Session ID`
+* `Peer Exists`
+
+#### 2. Get Contacts
+
+Get the contacts list of the user in the specified session.
+
+*Parameters*
+
+* `Session ID`: [String] Unique ID of the session.
+
+*Return*
+
+* `Contacts`: [List of Peer]
+
+*Errors*
+
+* `Unknown Session ID`
+
+#### 3. Open Payment Channel
+
+Open a payment channel with the peer with the specified initial balance.
+
+*Parameters*
+
+* `Session ID`: [String] Unique ID of the session.
+* `Peer Alias`: [String] Alias of the peer with whom channel is to be opened.
+* `Initial Channel Balance`: [Balance Info]
+
+*Return*
+
+* `Success`: [Boolean]
+
+*Errors*
+
+* `Unknown Session ID`
+* `Peer Not Responding`
+
+#### 4. Get Payment Channels
+
+Get the list of all payment channels in the specified session, that are open
+for off-chain transactions.
+
+*Parameters*
+
+* `Session ID`: [String] Unique ID of the session.
+
+*Return*
+
+* `Open channels`: [List of channel]
+
+*Errors*
+
+* `Unknown Session ID`
+
+#### 5. Subscribe To Payment Channel Requests
+
+Subscribe to notifications on new incoming payment channel requests.  A
+subscription id will be returned immediately and it will be included in the
+notification sent to the user on incoming payment channel requests. 
```

Comment:
How is this working and where is the channel request coming to?
Or is this more or less "open" to the implementation, so that gRPC for example needs to specify a dedicated method/callback, which will be the same for a webhook implementation?
Same for the other subscription in a channel below.

Reply:
The specification itself tells that, 
1. Client should be able to register for specific updates. Eg. Incoming channel updates. and
2. Server should be able to send these updates to the client.

As you said, how these is done, is open to the implementation. Some possible ways are
* server streaming in gRPC
* web hooks
* persistent, bidirectional connection using websockets etc.,